main.rb

```
@@ -0,0 +1,40 @@
+require 'rubygems'
+require 'bundler/setup'
+
+def bubble_sort(arr)
+  swap = true
+  while swap
+    swap = false
+    (0..arr.length - 2).each do |i|
+      if arr[i] > arr[i + 1]
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        swap = true
+      end
+    end
+  end
+  arr
+end
+
+puts bubble_sort([200, 22, 32, 112, 345])
+
+# reacreating the same function using
+
+def bubble_sort_by(arr)
+  swap = true
+  while swap
+    swap = false
+    (0..arr.length - 2).each do |i|
+      x = yield(arr[i], arr[i + 1]) if block_given?
+      if x.positive?
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        swap = true
+      end
+    end
+  end
+  print arr
```

- Your `bubble_sort_by` method is printing the result instead of returning it, that's why the result of the test is nil because nothing is returned.